packages/generate-persisted-query-manifest/src/index.ts

```diff
@@ -540,35 +550,48 @@ export async function generatePersistedQueryManifest(
 
   maybeReportErrorsAndExit(uniq(sources.map((source) => source.file)));
 
-  const inMemoryCacheConfig: InMemoryCacheConfig = {};
-  // Note that the ability to pass `addTypename: false` will be removed in
-  // Apollo Client v4. When we upgrade this package to work with AC4 as well,
-  // we'll want to do a version check and throw an error if the version is at
-  // least 4.0.0.
+  const cacheConfig: InMemoryCacheConfig = {};
+
   if ("addTypename" in config) {
-    inMemoryCacheConfig.addTypename = config.addTypename;
+    if (clientVersion.startsWith("4")) {
+      console.warn(
+        "`addTypename` was removed in Apollo Client 4 and is ignored. Please remove this option from your config.",
+      );
+    } else {
+      // @ts-ignore
+      cacheConfig.addTypename = config.addTypename;
+    }
   }
   // Using createFragmentRegistry means our minimum AC version is 3.7. We can
   // probably go back to 3.2 (original createPersistedQueryLink) if we just
   // reimplement/copy the fragment registry code here.
   if (fragmentRegistry) {
-    inMemoryCacheConfig.fragments = fragmentRegistry;
+    cacheConfig.fragments = fragmentRegistry;
   }
-  const cache = new InMemoryCache(inMemoryCacheConfig);
   const manifestOperationIds = new Map<string, string>();
   const manifestOperations: PersistedQueryManifestOperation[] = [];
-  const clientConfig: Partial<ApolloClientOptions<any>> = {};
+  const clientConfig: Partial<ApolloClientOptions> = {};
 
   if (config.documentTransform) {
     clientConfig.documentTransform = config.documentTransform;
   }
 
+  try {
+    // @ts-ignore
+    const { LocalState } = await import("@apollo/client/local-state");
+    clientConfig.localState = new LocalState();
+  } catch (e) {
+    // this is a v3 client
+  }
```

Comment:
Just questioning generally - do we *need* `LocalState` at all? I think we're stripping all `@client` fields anyways, right?